File: ASP.NET/SurveyApp.Domain/Models/Survey.cs

using System;
using System.Collections.Generic;

namespace SurveyApp.Domain.Models
{
    public class Survey
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public string Status { get; set; } = "active"; // Can be "active", "draft", or "archived"
        public int ResponseCount { get; set; }
        public int CompletionRate { get; set; }
        public List<Question> Questions { get; set; } = new List<Question>();
        public DeliveryConfiguration? DeliveryConfig { get; set; }
        
        // Navigation property
        public List<SurveyResponse>? Responses { get; set; }
    }

    public class Question
    {
        public int Id { get; set; }
        public string Text { get; set; } = string.Empty;
        public string Type { get; set; } = string.Empty;
        public bool Required { get; set; } = true;
        public string Description { get; set; } = string.Empty;
        public List<string> Options { get; set; } = new List<string>();
        public QuestionSettings? Settings { get; set; }
        
        // Propiedad Title para compatibilidad con la versión React
        public string Title
        {
            get { return Text; }
            set { Text = value; }
        }
    }

    public class QuestionSettings
    {
        public int? Min { get; set; }
        public int? Max { get; set; }
    }

    public class DeliveryConfiguration
    {
        public string Type { get; set; } = "manual"; // Can be "manual", "scheduled", or "trigger"
        public List<string> EmailAddresses { get; set; } = new List<string>();
        public ScheduleSettings? Schedule { get; set; }
        public TriggerSettings? Trigger { get; set; }
    }

    public class ScheduleSettings
    {
        public string Frequency { get; set; } = "monthly"; // Can be "daily", "weekly", "monthly"
        public int DayOfMonth { get; set; } = 1;
        public int? DayOfWeek { get; set; }
        public string Time { get; set; } = "09:00";
        public DateTime? StartDate { get; set; }
    }

    public class TriggerSettings
    {
        public string Type { get; set; } = "ticket-closed"; 
        public int DelayHours { get; set; } = 24;
        public bool SendAutomatically { get; set; } = false;
    }
}

File: src/domain/models/Survey.ts

export interface SurveyQuestion {
  id: string;
  title: string;
  description?: string;
  type: string;
  required: boolean;
  options?: string[];
  settings?: {
    min?: number;
    max?: number;
  };
}

export interface Survey {
  id: string;
  title: string;
  description?: string;
  questions: SurveyQuestion[];
  createdAt: string;
  deliveryConfig?: DeliveryConfig;
}

export interface QuestionResponse {
  questionId: string;
  questionTitle: string;
  questionType: string;
  value: string | string[];
  isValid: boolean;
}

export interface SurveyResponse {
  id?: string;
  surveyId: string;
  respondentName: string;
  respondentEmail: string;
  respondentPhone?: string;
  respondentCompany?: string;
  submittedAt: string;
  answers: QuestionResponse[];
  isExistingClient?: boolean;
  existingClientId?: string;
  completionTime?: number;
}

export interface SurveyResponseSubmission {
  surveyId: string;
  respondentName: string;
  respondentEmail: string;
  respondentPhone?: string;
  respondentCompany?: string;
  answers: Record<string, string | string[]>;
  isExistingClient?: boolean;
  existingClientId?: string;
  submittedAt?: string;
}

export interface DeliveryConfig {
  type: 'manual' | 'scheduled' | 'triggered';
  emailAddresses: string[];
  schedule?: {
    frequency: 'daily' | 'weekly' | 'monthly';
    dayOfMonth?: number;
    dayOfWeek?: number;
    time: string;
  };
  trigger?: {
    type: string;
    delayHours: number;
    sendAutomatically: boolean;
  };
}

export interface SurveyStatistics {
  totalResponses: number;
  averageCompletionTime: number;
  completionRate: number;
  questionStats: {
    questionId: string;
    questionTitle: string;
    responses: {
      answer: string;
      count: number;
      percentage: number;
    }[];
  }[];
}
